src/core/mapping/base_mapper.cc

```diff
@@ -682,9 +683,34 @@ bool BaseMapper::map_legate_store(const MapperContext ctx,
   // Generate layout constraints from the store mapping
   LayoutConstraintSet layout_constraints;
   mapping.populate_layout_constraints(layout_constraints);
+  auto& fields = layout_constraints.field_constraint.field_set;
+
+  // We need to hold the instance manager lock as we're about to try to find an instance
+  AutoLock lock(ctx, local_instances->manager_lock());
```

Comment:
Are you sure this is the right manager you want to lock? in the following you use `reduction_instances` if you're creating a reduction instance.

Reply:
you a right, fixing this